sklearn/model_selection/_validation.py

```diff
@@ -185,6 +192,10 @@ def cross_validate(estimator, X, y=None, *, groups=None, scoring=None, cv=None,
                 The estimator objects for each cv split.
                 This is available only if ``return_estimator`` parameter
                 is set to ``True``.
+            ``predictions``
+                Cross-validation predictions for the dataset.
+                This is available only if ``return_predictions`` parameter
+                is set to ``predict`` or ``predict_proba``.
 
```

Comment:
Maybe note the key 'test indices' too ?

Reply:
'test_indices' is being passed from `_fit_and_score` to `cross_validate` but not being returned by `cross_validate`. Do you think it is worth returning 'test_indices' from cross_validate?

Reply:
On second thought, I do think this could be useful to the user.

It also raises a follow-on question about the return array. Currently the predictions are flattened out to a 1D-array for 'predict' (ignoring 'predict_proba''s extra dimension for now). The sample predictions are ordered back into their positioning from (n_splits, validation_samples_per_split,) to (n_train_samples,) which loses information on which sample was used in which split.

Would it be better to return predictions and indices by the split (n_splits, validation_samples_per_split,)?

Reply:
@okz12 How about a "split_index"/"fold"/... key in the output with the fold index per sample. This would make it somewhat easy to subset the predictions by split. Having the predictions in the right order seems preferable?